rbac/default-role-manager/role_manager.go

```diff
@@ -176,9 +195,9 @@ func (rm *RoleManager) HasLink(name1 string, name2 string, domain ...string) (bo
 
 		roles, _ := rm.allDomains.LoadOrStore(domain, new(Roles))
 		allRoles = roles.(*Roles)
-		if !allRoles.hasRole(name1, rm.matchingFunc) || !allRoles.hasRole(name2, rm.matchingFunc) {
-			continue
-		}
+		//if !allRoles.hasRole(name1, rm.matchingFunc) || !allRoles.hasRole(name2, rm.matchingFunc) {
+		//	continue
+		//}
```

Comment:
why?

Reply:
if we support cross domain role inheriting, the two roles are not in the same domain